jlh/WeatherApp/app/src/main/java/com/e_conomic/weatherapp/domain/mappers/ForecastDataMapper.kt

```diff
@@ -0,0 +1,33 @@
+package com.e_conomic.weatherapp.domain.mappers
+
+import com.e_conomic.weatherapp.data.Forecast
+import com.e_conomic.weatherapp.data.ForecastResult
+import com.e_conomic.weatherapp.domain.model.ForecastList
+import java.text.DateFormat
+import java.util.*
+import com.e_conomic.weatherapp.domain.model.Forecast as ModelForecast
+
+class ForecastDataMapper {
+
+    fun convertFromDataModel(forecast: ForecastResult): ForecastList {
+        return ForecastList(forecast.city.name, forecast.city.country,
+                convertForecastListToDomain(forecast.list))
+    }
+
+    private fun convertForecastListToDomain(forecastList: List<Forecast>): List<ModelForecast> {
+        return forecastList.map { convertForecastItemToDomain(it) }
+    }
+
+    private fun convertForecastItemToDomain(forecast: Forecast): ModelForecast {
+        return ModelForecast(convertDate(forecast.dt), forecast.weather[0].description,
+                forecast.temp.max.toInt(), forecast.temp.min.toInt(),
+                generateIconUrl(forecast.weather[0].icon))
+    }
+
+    private fun generateIconUrl(iconCode: String) = "http://openweathermap.org/img/w/$iconCode.png"
+
+    private fun convertDate(date: Long): String {
+        val df = DateFormat.getDateInstance(DateFormat.MEDIUM, Locale.getDefault())
+        return df.format(date * 1000)
```

Comment:
Because the date you get from the API is in [Unix time](https://en.wikipedia.org/wiki/Unix_time)